src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -222,6 +221,43 @@ trait EntityComponent {
       entityAttributeQuery.batchInsertAttributes(attributeRecsToEntityId.keys.toSeq)
     }
 
+    private def upsertAttributes(entities: Traversable[Entity], entityRecs: Traversable[EntityRecord], referencedEntityRecs: Traversable[EntityRecord]) = {
+      def createEntityAttributeTempTable() = {
+        sql"""create temporary table if not exists ENTITY_ATTRIBUTE_TEMP (id bigint(20) unsigned NOT NULL AUTO_INCREMENT primary key, name text NOT NULL, value_string text, value_number double DEFAULT NULL, value_boolean bit(1) DEFAULT NULL, value_entity_ref bigint(20) unsigned DEFAULT NULL, list_index int(11) DEFAULT NULL, owner_id bigint(20) unsigned NOT NULL)""".as[Int]
+      }
+
+      def clearEntityAttributeTempTable() = {
+        sql"""delete from ENTITY_ATTRIBUTE_TEMP""".as[Int]
+      }
+
+      def insertTempAttributes(): WriteAction[Unit] = {
+        val entityIdsByName = referencedEntityRecs.map(r => AttributeEntityReference(r.entityType, r.name) -> r.id).toMap
```

Comment:
rename this val please, something to indicate that these are the referenced entites